plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergWritableTableHandle.java

```diff
@@ -85,9 +89,17 @@ public String getPartitionSpecAsJson()
     }
 
     @JsonProperty
-    public String getOutputPath()
+    public byte[] getSerializedLocationProvider()
+    {
+        return serializedLocationProvider;
+    }
+
+    public LocationProvider getLocationProvider()
     {
-        return outputPath;
+        if (locationProvider == null) {
+            locationProvider = deserializeFromBytes(serializedLocationProvider);
```

Comment:
We do not use Java serialization. In particular, it can lead to security issues.

Reply:
Sure, there are multiple ways we can go with this. We can also get the location provider through the dependency injection of table operation, since there is a general agreement that `HiveTableOperations` will be used across the board.

Reply:
Thought about it more, it would be inefficient to initialize a table operation, read table metadata and get the location provider. The alternative way is to pass in table properties. Please see if the new version works, thanks!